web/components/feed/feed-comments.tsx

```diff
@@ -256,19 +273,116 @@ export function FeedComment(props: {
           moreHref={contractPath(contract)}
           shouldTruncate={truncate}
         />
-        {onReplyClick && (
-          <button
-            className={'text-xs font-bold text-gray-500 hover:underline'}
-            onClick={() => onReplyClick(comment)}
-          >
-            Reply
-          </button>
-        )}
+        <Row className="mt-2 gap-4 text-gray-500">
+          <Tipper contract={contract} comment={comment} />
+
+          {onReplyClick && (
+            <button
+              className={'text-xs font-bold  hover:underline'}
+              onClick={() => onReplyClick(comment)}
+            >
+              Reply
+            </button>
+          )}
+        </Row>
       </div>
     </Row>
   )
 }
 
+function Tipper(prop: { contract: Contract; comment: Comment }) {
+  const { contract, comment } = prop
+  const { tips = {} } = comment
+
+  const me = useUser()
+  const myId = me?.id || ''
+  const savedTip = tips[myId] || 0
+
+  // optimistically increase the tip count, but debounce the update
+  const [localTip, setLocalTip] = useState(savedTip)
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const score = useMemo(
+    flow(
```

Comment:
It's cool, and I didn't know lodash had this, but I think it would be clearer to write `const tipValues = Object.values(tips).concat(localTip); return sumBy(tipValues, invQuad)`.

Reply:
mqp I agree, with the caveat that we need the `{ ...tips, [me]: localTip } thing to override your stored tip otherwise you'll double count your own tipped contribution if any of it is stored already.

ofc this is moot if we just do it in the txns